internal/multicluster/internal/controllers/exportedservices/controller.go

```diff
@@ -293,3 +298,43 @@ func sortRefValue(m map[resource.ReferenceKey]*serviceExports) []*serviceExports
 	})
 	return vals
 }
+
+func getNamespaceForServices(exportedServices []*resource.DecodedResource[*pbmulticluster.ExportedServices], namespaceExportedServices []*resource.DecodedResource[*pbmulticluster.NamespaceExportedServices], partitionedExportedServices []*resource.DecodedResource[*pbmulticluster.PartitionExportedServices]) string {
```

Comment:
aside: you may find it clearer to replicate this type alias trick in `internal/multicluster/internal/types/decoded.go` to shorten `*resource.DecodedResource[*pbmulticluster.XXX]` down to `*resource.DecodedXXX`

https://github.com/hashicorp/consul/blob/main/internal/mesh/internal/types/decoded.go

Reply:
thnx for this. have made this change